lib/shelf_proxy.dart

```diff
@@ -21,66 +21,65 @@ import 'src/utils.dart';
 /// [proxyName] is used in headers to identify this proxy. It should be a valid
 /// HTTP token or a hostname. It defaults to `shelf_proxy`.
 Handler proxyHandler(url, {http.Client client, String proxyName}) {
-  if (url is String) url = Uri.parse(url);
-  if (client == null) client = new http.Client();
-  if (proxyName == null) proxyName = 'shelf_proxy';
+  var uri = (url is String) ? Uri.parse(url) : url as Uri;
```

Comment:
We should throw an `ArgumentError` here if `url` isn't a `String` or `Uri`.

Nit: unnecessary parens